celements-spring-security/src/main/java/com/celements/spring/security/WikiClientRegistrationRepository.java

```diff
@@ -0,0 +1,43 @@
+package com.celements.spring.security;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.oidc.IdTokenClaimNames;
+import org.springframework.stereotype.Component;
+
+@Component
+public class WikiClientRegistrationRepository implements ClientRegistrationRepository {
+
+  private static final Logger LOGGER = LoggerFactory
+      .getLogger(WikiClientRegistrationRepository.class);
+
+  private final IdentityService identityService;
+
+  @Inject
+  public WikiClientRegistrationRepository(IdentityService identityService) {
+    this.identityService = identityService;
+  }
+
+  @Override
+  public ClientRegistration findByRegistrationId(String registrationId) {
+    LOGGER.debug("findByRegistrationId for {}", registrationId);
+    return ClientRegistration.withRegistrationId(registrationId)
```

Comment:
perhaps validate that given `registrationId` actually matches `identityService.getRegistrationId()` and fast fail if not. otherwise mismatching `ClientRegistration` data may be returned.